topics/tutorials/file_and_code_templates/using_file_templates.md

```diff
@@ -0,0 +1,127 @@
+File templates assigned to the <control>[Files](providing_file_templates.md#files)</control> category are automatically available in the <menupath>File | New</menupath> action group.
+Sometimes, creating a file from a given template in a specific project place doesn't make sense, or a template requires some additional properties for its content.
+It is possible to control a file template's visibility and its available properties using
+[`CreateFromTemplateHandler`](upsource:///platform/lang-impl/src/com/intellij/ide/fileTemplates/CreateFromTemplateHandler.java)
+implementation registered in the `com.intellij.createFromTemplateHandler` EP.
+
+**Example**:
+[`JavaCreateFromTemplateHandler`](upsource:///java/java-impl/src/com/intellij/ide/fileTemplates/JavaCreateFromTemplateHandler.java)
+
+### Exposing File Templates from the Other Category
+
+File templates from the <control>[Other](providing_file_templates.md#other)</control> category are not exposed by default.
+To make them available in the UI, a plugin has to implement and register an action.
+The easiest way to do it is by extending
+[`CreateFileFromTemplateAction`](upsource:///platform/lang-impl/src/com/intellij/ide/actions/CreateFileFromTemplateAction.java)
+base class which contains standard implementations of necessary methods.
+It allows customizing the action by providing an action name, icon, overriding methods creating dialog, and others.
+Example action:
+
+```java
+public class CreateMyClassAction extends CreateFileFromTemplateAction {
+
+  public CreateMyClassAction() {
+    super("My Class", "Creates new class", CLASS_ICON);
+  }
+
+  @Override
+  protected void buildDialog(Project project, PsiDirectory directory,
+      CreateFileFromTemplateDialog.Builder builder) {
+    builder
+    .setTitle("New My File")
+    .addKind("Class", CLASS_ICON, "My Class");
+  }
+
+  @Override
+  protected String getActionName(PsiDirectory directory,
+      @NotNull String newName, String templateName) {
+    return "Create My Class: " + newName;
+  }
+}
+```
+
+The new action should be registered under the `NewGroup` group, e.g:
+
+```xml
+<actions>
+  <action id="Create.MyClass" class="com.example.CreateMyClassAction">
+    <add-to-group group-id="NewGroup"/>
+  </action>
+</actions>
+```
+
+### Custom "Create File From Template" Actions
+
+In some cases, the default mechanism for creating files from templates is insufficient.
+Consider a language that defines multiple types of core entities, e.g., in the Java language, the following entities can be created: Class, Interface, Record, Enum, and Annotation.
+
+Having all of those items in the <menupath>File | New</menupath> action group may overwhelm users with the number of options to choose.
+It is more user-friendly to provide a single <menupath>File | New | Java Class</menupath> action and let users choose a specific entity type in the creation dialog:
+
+![Create Java class](new_java_class_popup.png){width="330"}
```

Comment:
Style guide says

> For consistency reasons, the images must be 296, 460, or 706 pixels wide.

I'd also add a border so that the white doesn't "bleed" into the white of the page background. 

```
![Create Java class](new_java_class_popup.png){width="460" border-effect="line"}
```

Reply:
Good point, thanks, fixed.